sematic/ee/plugins/external_resource/ray/load_testing/__main__.py

```diff
@@ -0,0 +1,208 @@
+    )
+    parser.add_argument(
+        "--collatz.n_workers",
+        type=int,
+        default=0,
+        dest="collatz_n_workers",
+        help=COLLATZ_N_WORKERS_HELP,
+    )
+    parser.add_argument(
+        "--collatz.n_tasks",
+        type=int,
+        default=10,
+        dest="collatz_n_tasks",
+        help=COLLATZ_N_TASKS_HELP,
+    )
+    parser.add_argument(
+        "--collatz.wait_minutes",
+        type=int,
+        default=1,
+        dest="collatz_wait_minutes",
+        help=COLLATZ_WAIT_MINUTES_HELP,
+    )
+    parser.add_argument(
+        "--collatz.memory_growth_factor",
+        type=float,
+        default=1.10,
+        dest="collatz_memory_growth_factor",
+        help=COLLATZ_MEMORY_GROWTH_FACTOR_HELP,
+    )
+
+    parser.add_argument(
+        "--mnist.n_rates",
+        type=int,
+        default=10,
+        dest="mnist_n_rates",
+        help=MNIST_N_RATES_HELP,
+    )
+    parser.add_argument(
+        "--mnist.n_epochs",
+        type=int,
+        default=10,
+        dest="mnist_n_epochs",
+        help=MNIST_N_EPOCHS_HELP,
+    )
+    parser.add_argument(
+        "--mnist.n_workers",
+        type=int,
+        default=0,
+        dest="mnist_n_workers",
+        help=MNIST_N_WORKERS_HELP,
+    )
+    parser.add_argument(
+        "--mnist.wait_minutes",
+        type=int,
+        default=10,
+        dest="mnist_wait_minutes",
+        help=MNIST_WAIT_MINUTES_HELP,
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def _get_resolver(args: argparse.Namespace) -> Resolver:
+    """Instantiates the Resolver based on the passed arguments."""
+    if args.silent:
+        return SilentResolver()
+    if not args.cloud:
+        return LocalResolver()
+
+    return CloudResolver(
+        detach=True,
+    )
+
+
+def _wait_for_debugger():
+    debugpy.listen(5724)
+
+    print("Waiting for debugger to attach...")
+    debugpy.wait_for_client()
+    print("Debugger attached")
+
+
+def main() -> None:
+    if os.environ.get("DEBUGPY", None) is not None:
+        _wait_for_debugger()
+
+    args = _parse_args()
+    logger.info("Command line arguments: %s", args)
+    tags = ["load-testing", "ray"]
+
+    collatz_config_args = {
+        arg_key.replace("collatz_", ""): arg_val
+        for arg_key, arg_val in vars(args).items()
+        if arg_key.startswith("collatz_")
+    }
+    collatz_config: Optional[CollatzConfig] = CollatzConfig(**collatz_config_args)
+    if collatz_config.n_workers == 0:  # type: ignore
+        collatz_config = None
+    else:
+        tags.append("collatz-tested")
+
+    mnist_config_args = {
+        arg_key.replace("mnist_", ""): arg_val
+        for arg_key, arg_val in vars(args).items()
+        if arg_key.startswith("mnist_")
+    }
+    mnist_config: Optional[MnistConfig] = MnistConfig(**mnist_config_args)
+    if mnist_config.n_workers == 0:  # type: ignore
+        mnist_config = None
+    else:
+        tags.append("mnist-tested")
```

Comment:
Please do all this arg processing inside `_parse_args`. You can return the `CollatzConfig` and the `MnistConfig` under new, specific keys.